src/components/Countries.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getAllCountries } from '../redux/countries';
+import Country from './Country';
+import numberFormater from './FormatNumber';
+
+const Countries = () => {
+  const countries = useSelector((state) => state.countries.countries);
+  const data = useSelector((state) => state.countries.data);
+  const dispatch = useDispatch();
+  const [totalConfirmed, setTotalConfirmed] = useState(0);
+  const [totalDeaths, setTotalDeaths] = useState(0);
+
+  useEffect(() => {
+    dispatch(getAllCountries());
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (data && data !== []) {
```

Comment:
- [ ] <b>REQUIRED:</b>  This condition will always return true. that is because Arrays or Object in javascript do not have referential equality to each other. that's because they are not premitive values, so `[ ] === [ ] or { } === { }` even if they seem to be equal but that condition will always return false. so your condition will always return true, even if the data are actually the same. 
To enforce the condition you wanted to enforce, you can use `shallowEqual` instead. which will help your component to only re-render when it is necessary. This is because useSelector will always force your component to re-render every time the state changes, and to avoid that, you must make sure that useSelector will re-render the component if and only if the previous data are different from the new data in the state(change here can be even one property on one of the countries objects), and since the arrays and objects are not primitive values, you can't use referential equality(`===` Because it will always return false or always return true when is `!==`). so, that is where  `shallowEqual` comes to the rescue. 

```jsx
import { useSelector, shallowEqual } from 'react-redux';

// And the you use it like this
const countries = useSelector((state) => state.countries.countries, shallowEqual);
const data = useSelector((state) => state.countries.data, shallowEqual);
```